vizro-core/docs/pages/user-guides/data.md

```diff
@@ -268,3 +265,81 @@ data_manager["slow_expire_data"].timeout = 60 * 60
+- give an `id` to all components that have the data source you wish to alter through a parameter.
+- [add a parameter](parameters.md) with `targets` of the form `<target_component_id>.data_frame.<dynamic_data_argument>` and a suitable [selector](selectors.md).
+
+For example, let us extend the [dynamic data example](#dynamic-data) above to show how the `load_iris_data` can take an argument `number_of_points` controlled from the dashboard with a [`Slider`][vizro.models.Slider].
+
+!!! example "Parametrized dynamic data"
+    === "app.py"
+        ```py hl_lines="8 10 20-23"
+        from vizro import Vizro
+        import pandas as pd
+        import vizro.plotly.express as px
+        import vizro.models as vm
+
+        from vizro.managers import data_manager
+
+        def load_iris_data(number_of_points=10): # (1)!
+            iris = pd.read_csv("iris.csv") # (2)!
+            return iris.sample(number_of_points) # (3)!
+
+        data_manager["iris"] = load_iris_data # (4)!
+
+        page = vm.Page(
+            title="Update the chart on page refresh",
+            components=[
+                vm.Graph(id="graph", figure=px.box("iris", x="species", y="petal_width", color="species")) # (5)!
+            ],
+            controls=[
+                vm.Parameter(
+                    targets=["graph.data_frame.number_of_points"], # (6)!
+                    selector=vm.Slider(min=1, max=20, step=1, value=10),
+                )
+            ],
+        )
+
+        dashboard = vm.Dashboard(pages=[page])
+
+        Vizro().build(dashboard).run()
+        ```
+
+        1. `load_iris_data` now takes a single argument, `number_of_points`, with a default value of 10.
+        2. `iris` is still a pandas DataFrame created by reading from the CSV file `iris.csv`.
+        3. Sample points at random, where `number_of_points` gives the number of points selected.
+        4. To use `load_iris_data` as dynamic data it must be added to the data manager. You should **not** actually call the function as `load_iris_data()` or `load_iris_data(number_of_points=...)`; doing so would result in static data that cannot be reloaded.
+        5. Give the `vm.Graph` component `id="graph"` so that the `vm.Parameter` can target it. Dynamic data is referenced by the name of the data source `"iris"`.
+        6. Create a `vm.Parameter` to target the `number_of_points` argument for the `data_frame` used in `graph`.
+
+    === "Result"
+        [![ParametrizedDynamicData]][ParametrizedDynamicData]
+
+    [ParametrizedDynamicData]: ../../assets/user_guides/data/parametrized_dynamic_data.gif
+
+Parametrized data loading is compatible with [caching](#configure-cache). The cache uses [memoization](https://flask-caching.readthedocs.io/en/latest/#memoization), so that the dynamic data function's arguments are included in the cache key. This means that, for example `load_iris_data(number_of_points=10)` is cached independently of `load_iris_data(number_of_points=20)`.
```

Comment:
```suggestion
Parametrized data loading is compatible with [caching](#configure-cache). The cache uses [memoization](https://flask-caching.readthedocs.io/en/latest/#memoization), so that the dynamic data function's arguments are included in the cache key. This means that `load_iris_data(number_of_points=10)` is cached independently of `load_iris_data(number_of_points=20)`.
```